task-sdk/src/airflow/sdk/definitions/asset/decorators.py

```diff
@@ -45,7 +45,7 @@ def __init__(self, *, definition_name: str, uri: str | None = None, **kwargs) ->
     @classmethod
     def from_definition(cls, definition: AssetDefinition | MultiAssetDefinition) -> Self:
         return cls(
-            task_id="__main__",
+            task_id=definition._function.__name__,
```

Comment:
I feel we probably should use 
```suggestion
            task_id=definition.name,
```
instead?

Reply:
So that means assent name becomes the task_id, so its more or like question now , which one is better serves here. If user removes the asset name, can we consider that asset is invalid? in this case a new task will appear with the function name. as there is no asset name it picks up function name as asset name.

Reply:
>  If user removes the asset name, can we consider that asset is invalid

it depends. The asset might still exist elsewhere. So I guess the answer to this one is no.

> in this case a new task will appear with the function name. as there is no asset name it picks up function name as asset name.

yep, just like the dag itself

Reply:
okay, thanks updated :)